constraints_test.go

```diff
@@ -166,11 +166,28 @@ func TestConstraintCheck(t *testing.T) {
 		{"<=1.1", "0.1.0", true},
 		{"<=1.1", "1.1.0", true},
 		{"<=1.1", "1.1.1", false},
+		{"<=1.1-alpha1", "1.1", false},
+		{"<=2.x", "3.0.0", false},
+		{"<=2.x", "2.9.9", true},
+		{"<2.x", "2.0.0", false},
```

Comment:
yeah, i think that's what i did when i first wrote this, too. it was a PITA writing this patch, too, i had to whiteboard it out to keep them from mushing in my head 😄